src/main/java/labs/introtoprogramming/lab5/scene/SceneObject.java

```diff
@@ -37,4 +41,12 @@ public Mesh getMesh() {
   public boolean intersect(Ray ray) {
     return false;
   }
+
+  public void setInput(Input input) {
```

Comment:
Most of object are static and this functionality is redundant for them. Maybe is it more reasonable to get data from input and than apply it to object via transform?